### ISSUE TITLE
linter: `--print-config` does not expand category rules for plugins inside overrides

### ISSUE THREAD

**Issue by sonion028** [oxc-project/oxc#24878]
### What version of Oxlint are you using?

`1.75.0`

### What command did you run?

```sh
oxlint --print-config src/repro.ts
oxlint --format json src/repro.ts --no-error-on-unmatched-pattern
```

### What does your `.oxlintrc.json` config file look like?

```json
{
  "categories": {
    "correctness": "error"
  },
  "overrides": [
    {
      "files": ["src/**/*.ts"],
      "plugins": ["jsdoc"]
    }
  ]
}
```

### What source file did you lint?

`src/repro.ts`

```ts
/**
 * @implements {Foo}
 */
function notAClass() {}
notAClass();
```

### What happened?

`--print-config` prints the override plugin, but it does not expand any rules enabled by the root `categories.correctness` setting for that override plugin.

Relevant part of `oxlint --print-config src/repro.ts`:

```json
{
  "plugins": ["unicorn", "typescript", "oxc"],
  "categories": {
    "correctness": "deny"
  },
  "rules": {
    "...": "root correctness rules for default/root plugins are expanded here"
  },
  "overrides": [
    {
      "files": ["src/**/*.ts"],
      "plugins": ["jsdoc"],
      "rules": {}
    }
  ]
}
```

`jsdoc/implements-on-classes` is missing from the printed config.

However, normal linting proves the rule is actually enabled and effective:

```json
{
  "diagnostics": [
    {
      "message": "`@implements` used on a non-constructor function",
      "code": "jsdoc(implements-on-classes)",
      "severity": "error"
    }
  ],
  "number_of_files": 1,
  "number_of_rules": 95
}
```

I also checked the root-plugin case for comparison:

```json
{
  "plugins": ["jsdoc"],
  "categories": {
    "correctness": "error"
  }
}
```

For the root-plugin case, `oxlint --print-config src/repro.ts` does include `jsdoc/implements-on-classes: "deny"`. So this seems specific to `plugins` declared inside `overrides` combined with category-enabled rules.

### What did you expect to happen?

`--print-config` should print the final effective configuration for the target file.

For the repro above, because `oxlint --format json src/repro.ts` actually runs `jsdoc(implements-on-classes)`, I expected `--print-config src/repro.ts` to include the effective JSDoc correctness rules enabled by:

- root `categories.correctness = "error"`
- override `plugins = ["jsdoc"]`
- target file matching `src/**/*.ts`

For example, `jsdoc/implements-on-classes` should appear as `"deny"` in the printed effective rules for `src/repro.ts`.

### Why this matters

The CLI docs say `--print-config` “outputs the configuration to be used”. I could not find another command that prints or verifies the final effective config for a target file.

If `--print-config` omits category-expanded rules for override plugins, there is no reliable way to inspect the final effective Oxlint configuration during migration/debugging. Running lint probes can prove that a specific rule fires, but it is not a replacement for inspecting the complete effective configuration.

Related but not the same:

- #22117 is about `jsPlugins` rules being dropped from `--print-config` output.
- #17724 is about expanding/resolving extended configs.

This issue is about built-in Oxlint plugins (for example `jsdoc`) declared in `overrides[].plugins` and enabled via root `categories`.
